IPython/core/debugger.py

```diff
@@ -621,3 +621,13 @@ def do_where(self, arg):
                 self.print_stack_trace()
 
         do_w = do_where
+
+
+def set_trace(frame=None):
```

Comment:
Happy to add it to the function, it should likely be passed to `Pdb.set_trace()` by adding 1 to it, and we should even likely start the debugging 1 frame up from the current one to be consistent with normal Pdb.